common/src/main/java/tachyon/resource/ResourcePool.java

```diff
@@ -66,19 +67,33 @@ protected ResourcePool(int maxCapacity, BlockingQueue<T> resources) {
    */
   public T acquire() {
     // If the resource pool is empty but capacity is not yet full, create a new resource.
-    synchronized (mCapacityLock) {
-      if (mResources.isEmpty() && mCurrentCapacity < mMaxCapacity) {
-        T newResource = createNewResource();
-        mCurrentCapacity ++;
-        return newResource;
-      }
+    T resource = mResources.poll();
+    if (resource != null) {
+      return resource;
+    }
+
+    if (mCurrentCapacity.getAndIncrement() < mMaxCapacity) {
```

Comment:
I may be mistaken, but I think it might have to be "<". 

If the capacity is 1, and current capacity would start at 0. Then the first getAndIncrement() would return 0. The second getAndIncrement() would return 1. If the condition were "<=", then I think 2 new resources would be created?